Typecheck failed: `@submit` of `<AuthForm>` `<Form>` , `Target signature provides too few arguments`

### Environment

|                      |                                                                 |
| -------------------- | --------------------------------------------------------------- |
| **Operating system** | `Windows 10.0.26200`                                            |
| **CPU**              | `Intel(R) Core(TM) Ultra 5 225H (14 cores)`                     |
| **Node.js version**  | `v24.18.0`                                                      |
| **nuxt/cli version** | `3.37.0`                                                        |
| **Package manager**  | `bun@1.3.14`                                                    |
| **Nuxt version**     | `4.5.2`                                                         |
| **Nitro version**    | `2.13.4`                                                        |
| **Builder**          | `vite@8.2.1`                                                    |
| **Config**           | `app`, `compatibilityDate`, `css`, `fonts`, `icon`, `modules`   |
| **Modules**          | `@nuxt/ui@4.10.0`                                               |


### Is this bug related to Nuxt or Vue?

Nuxt

### Package

v4.x

### Version

v4.10.0

### Reproduction

N/A - it works in [StackBlitz](https://stackblitz.com/edit/nuxt-demo-e8b3297?file=app%2Fpages%2Findex.vue)? idk

### Description

```
app/pages/login.vue:70:8 - error TS2322: Type '(payload: FormSubmitEvent<{ email: string; password: string; rememberMe: boolean; }>) => Promise<void>' is not assignable to type '(((() => void) | ((event: FormSubmitEvent<{ email: string; password: string; rememberMe: boolean; }>) => void)) & ((payload: FormSubmitEvent<{ email: string; password: string; rememberMe?: boolean | undefined; }>) => any)) | undefined'.
  Type '(payload: FormSubmitEvent<{ email: string; password: string; rememberMe: boolean; }>) => Promise<void>' is not assignable to type '(() => void) & ((payload: FormSubmitEvent<{ email: string; password: string; rememberMe?: boolean | undefined; }>) => any)'.
    Type '(payload: FormSubmitEvent<{ email: string; password: string; rememberMe: boolean; }>) => Promise<void>' is not assignable to type '() => void'.
      Target signature provides too few arguments. Expected 1 or more, but got 0.

70       @submit="onSubmit"
          ~~~~~~

  node_modules/.bun/@nuxt+ui@4.10.0+53c4fa4b58ab7d90/node_modules/@nuxt/ui/dist/runtime/components/AuthForm.d.vue.ts:80:5
    80     onSubmit?: FormProps<T>['onSubmit'];
           ~~~~~~~~
    The expected type comes from property 'onSubmit' which is declared here on type 'NonNullable<VNodeProps & AllowedComponentProps & ComponentCustomProps & { as?: any; icon?: any; title?: string | undefined; description?: string | undefined; ... 21 more ...; novalidate?: Booleanish | undefined; } & Record<...>>'
```

https://npmx.dev/package-code/@nuxt/ui/v/4.10.0/dist%2Fruntime%2Fcomponents%2FForm.d.vue.ts#L54

> ```ts
>     onSubmit?: ((event: FormSubmitEvent<FormData<S, T>>) => void) | (() => void);
> //                                                                  ^^^^^^^^^^^^
> ```

### Additional context

_No response_

### Logs

```shell-script

```

--- a/src/runtime/components/AuthForm.vue
+++ b/src/runtime/components/AuthForm.vue
@@ -13,7 +13,7 @@ import type { SelectMenuProps } from './SelectMenu.vue'
 import type { PinInputProps } from './PinInput.vue'
 import type { IconProps } from './Icon.vue'
 import type { LinkPropsKeys } from './Link.vue'
-import type { FormSchema, FormSubmitEvent, InferInput } from '../types/form'
+import type { FormData, FormSchema, FormSubmitEvent, InferInput } from '../types/form'
 import type { FormHTMLAttributes } from '../types/html'
 import type { NonUnion } from '../types/utils'
 import type { ComponentConfig } from '../types/tv'
@@ -151,7 +151,7 @@ const state = reactive<FormStateType>((_props.fields as TypedAuthFormField[] || 
   return acc
 }, {} as FormStateType))
 
-defineEmits<AuthFormEmits<typeof state>>()
+defineEmits<AuthFormEmits<FormData<T>>>()
 const slots = defineSlots<AuthFormSlots<typeof state, F>>()
 
 const props = useComponentProps<AuthFormProps<T, F>>('authForm', _props)

--- a/test/components/AuthForm.spec.ts
+++ b/test/components/AuthForm.spec.ts
@@ -1,10 +1,12 @@
-import { describe, it, expect } from 'vitest'
+import { describe, it, expect, test } from 'vitest'
 import { axe } from 'vitest-axe'
 import { mountSuspended } from '@nuxt/test-utils/runtime'
+import * as z from 'zod'
 import AuthForm from '../../src/runtime/components/AuthForm.vue'
-import type { FormSchema } from '../../src/runtime/types/form'
+import type { FormSchema, FormSubmitEvent } from '../../src/runtime/types/form'
 import type { AuthFormProps } from '../../src/runtime/components/AuthForm.vue'
 import { renderEach } from '../component-render'
+import { expectEmitPayloadType } from '../utils/types'
 
 describe('AuthForm', () => {
   const fields = [{
@@ -89,5 +91,18 @@ describe('AuthForm', () => {
     })
 
     expect(await axe(wrapper.element)).toHaveNoViolations()
+  })
+
+  test('should have the correct types', () => {
+    const schema = z.object({
+      email: z.string(),
+      rememberMe: z.boolean().default(false)
+    })
+
+    // with a schema default
+    expectEmitPayloadType('submit', () => AuthForm({
+      schema,
+      fields: [{ name: 'email', type: 'text' }, { name: 'rememberMe', type: 'checkbox' }]
+    })).toEqualTypeOf<[FormSubmitEvent<{ email: string, rememberMe: boolean }>]>()
   })
 })

